File: tests/fixtures/extract/sample.rb
module Demo
  class Store
    def merge(items)
      added = 0
      items.each do |key, value|
        unless @items.key?(key)
          @items[key] = value
          added += 1
        end
      end
      added
    end
  end

  def self.collect_names(rows)
    normalize = ->(row) do
      name = row[:name].to_s.strip
      if name.empty?
        nil
      else
        name
      end
    end
    rows.filter_map do |row|
      normalize.call(row)
    end
  end
end

File: tests/fixtures/extract/sample.php
<?php

namespace Demo;

class Store
{
    public function merge(array $items): int
    {
        $added = 0;
        foreach ($items as $key => $value) {
            if (!array_key_exists($key, $this->items)) {
                $this->items[$key] = $value;
                $added++;
            }
        }
        return $added;
    }
}

function collectNames(array $rows): array
{
    $normalize = function (array $row): ?string {
        if (!isset($row['name'])) {
            return null;
        }
        $name = trim($row['name']);
        return $name === '' ? null : $name;
    };
    $names = [];
    foreach ($rows as $row) {
        $name = $normalize($row);
        if ($name !== null) {
            $names[] = $name;
        }
    }
    return $names;
}
